### DISCUSSION TITLE
Export Template isn't showing custom fields

### DISCUSSION THREAD

**Discussion by sh83le** [netbox-community/netbox#14347] [Help Wanted!]
Dear all,

I created a simple Export Template, for exporting some prefixes.

I am not able to get the custom_fields value there.
I created some vars: mc_network_dhcpserver, mc_network_dnsserver1, mc_network_dnsserver2

After this I created a template, but only the default vars will be shown in the export.
I tried:
mc_network_dnsserver1, cf_mc_network_dnsserver1, prefix.mc_network_dnsserver1 etc. 
I got no value and there are values behind each field.

What is wrong in my export template ?

prefix,status,tenant,site,vlan,description,cf_mc_network_ipv4gateway,cf_mc_network_dhcpserver,cf_mc_network_dnsserver1,cf_mc_network_dnsserver2,cf_mc_network_dnsserver3,cf_mcstandort

```
{% for prefix in queryset %}
{{ prefix.prefix }},{{ prefix.status }},{{ prefix.tenant }},{{ prefix.site }},{{ prefix.vlan.vid if prefix.vlan else '' }},
{{ prefix.description }},{{ cf_mc_network_ipv4gateway }},
{{ cf_mc_network_dhcpserver}},
{{ prefix.cf_mc_network_dnsserver1 }},
{{ prefix.cf_mc_network_dnsserver2 }},
{{ cf_mc_network_dnsserver3 }},{{cf_mcstandort }}
{% endfor %}
```

**Comment by candlerb** [comment]
Try these. I think they should all work - not entirely sure about the first one but I think jinja2 allows it.

* `prefix.cf.mc_network_dhcpserver`
* `prefix.cf["mc_network_dhcpserver"]`
* `prefix.custom_field_data["mc_network_dhcpserver"]`

A good way to explore the attributes available on an object is to use [nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/):

```
/opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py nbshell

>>> d = Device.objects.get(name="gw1")
>>> d.cf["monitoring"]
['snmp']
>>> 
```
(Jinja2 isn't exactly like Python but is very close). `help(Device)` gives detailed info about that class.
